tests/test_ilastik/helpers/wait.py

```diff
@@ -0,0 +1,112 @@
+###############################################################################
+#   ilastik: interactive learning and segmentation toolkit
+#
+#       Copyright (C) 2011-2023, the ilastik developers
+#                                <team@ilastik.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# In addition, as a special exception, the copyright holders of
+# ilastik give you permission to combine ilastik with applets,
+# workflows and plugins which are not covered under the GNU
+# General Public License.
+#
+# See the LICENSE file for details. License information is also available
+# on the ilastik web site at:
+#          http://ilastik.org/license.html
+###############################################################################
+import contextlib
+import threading
+import time
+from typing import Optional
+
+from PyQt5.QtCore import QEventLoop, Qt, QTimer
+from PyQt5.QtWidgets import QApplication
+
+
+def wait_until(f, timeout=10):
+    result = f()
+    while not result and timeout:
+        timeout -= 1
+        time.sleep(1)
+        result = f()
+    if not result:
+        raise Exception(f"TIMEOUT! waiting for {f}")
+    return result
+
+
+@contextlib.contextmanager
+def wait_signal(signal, timeout=1000):
+    """
+    Context manager
+    on exit would wait for signal to fire before continuing
+    :param timeout: timeout in ms
+    :param signal:
+    """
+    evtLoop = QEventLoop()
+    # QueuedConnection so we don't receive signal before entering the loop
+    signal.connect(evtLoop.quit, Qt.QueuedConnection)
+
+    yield
+
+    _exec_with_timeout(evtLoop, timeout, signal)
```

Comment:
Wrap `yield` in `try...finally`?

Reply:
yeah will be nicer to handle quitting of loop in this function